File: ChatSTNR/app/src/main/java/com/example/chatstnr/SplashActivity.java
package com.example.chatstnr;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.example.chatstnr.utils.FirebaseUtil;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if(FirebaseUtil.isLoggedin()){
                    startActivity(new Intent(SplashActivity.this, MainActivity.class));
                }else{
                    startActivity(new Intent(SplashActivity.this, LoginActivity.class));
                }
                finish();
            }
        }, 3000);

    }
}

File: RahilChatApplication/app/src/main/java/com/example/rahilchatapplication/SplashActivity.kt
package com.example.rahilchatapplication

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash) // Set the layout for your splash screen
        supportActionBar?.hide()
        // Use a Handler to delay the transition to the main activity
        Handler().postDelayed({
            // Start your main activity
            val intent = Intent(this@SplashActivity, LogIn::class.java)
            startActivity(intent)
            finish() // close the splash screen activity
        }, 2000) // 2000 milliseconds (2 seconds) delay
    }
}
